contracts/examples/dao/DAO.sol

```diff
@@ -25,12 +25,13 @@ contract DAO is VentureEth {
     using EnumerableSet for EnumerableSet.AddressSet;
 
     event VentureProposed(address proposal);
+    event ProfitProposed(address proposal);
     event VentureAdded(address venture);
 
-    uint256 public investmentPool;
     uint256 public threshold;
 
-    mapping(address => address) private proposals;
+    mapping(address => address) private ventureProposals;
+    address private dividendProposal;
```

Comment:
Why do we need a separate variable for `dividendProposal`, can't we just have a mapping for all proposals, regardless of what they are about?

Reply:
Because `proposals` would have to have a single type. If you refer to `ventureProposals`, that would be `mapping (address => address)`. If you refer to dividends, then that would be `mapping(uint256=>address)`. But since amount quantities are fungible, thus not making sense to vote on more than one retrieval at a time, and since it would be impossible to distinguish between the amounts solely by their quantity, I reduced the latter mapping only to a single voting address, `dividendProposal`.

Reply:
I think that is overcomplicating the issue. You can have multiple simultaneous dividends proposals:
- "We've got lots of money, should we distribute dividends?"
- "I propose to distribute 1M ETH"
- "I propose to distribute 2M ETH instead"
And let people vote.

At the end, all that you need to know from a proposal is whether it was created from `DAO.sol`, and therefore allowed to execute functions, or not, in which case it doesn't have permissions.

Reply:
Yes, I think you're right, but still, you would need two mappings: `(address=>address)` and `(uint256=>address)`. And, I'm not sure amount would a good index, maybe`block.now`?